### PR TITLE
Biting with no teeth now "gums" people.

### PR DESCRIPTION
**Pull request by PrimeDSS13**
Small rework and feature addition.
Previously, it just put you back on punch intent when you had less than two teeth. Now you "gum" people, which does 1 damage consistent instead of the 1-5 random from a normal punch. Gives a feedback message when you can't suck blood as a vampire due to your lack of teeth. 

Weird thing I'd like to mention is that certain races, such as Insectoids, Vox, and Grays have no teeth in their butchercode, it's merely null when you VV it. That means surgery to remove teeth doesn't but, but I think you can still punch teeth out of them? It's weird.

:cl:
 * rscadd: When you have all your teeth removed, you "gum" people instead of just automatically switching to punch.
 * tweak: Vampires now get a message saying "You cannot suck blood with no teeth" Instead of punching their victims.

### CODE REVIEW

**Comment by boy2mantwicethefam** [comment]
I'm glad /vg/ is taking steps to make british people feel comfortable and accepted

File: code/modules/mob/living/carbon/human/human.dm [hunk 1]
```diff
@@ -1584,7 +1584,7 @@
 		return 1
 
 	var/datum/butchering_product/teeth/T = locate(/datum/butchering_product/teeth) in src.butchering_drops
-	if(T && T.amount >= 2)
+	if(T && T.amount >= -1)
 		return 1
 
 	return 0
```
Comment (gurfan) [inline code/modules/mob/living/carbon/human/human.dm:1590]: you can probably just replace lines 1586-1590 with just ```return 1``` if the amount of teeth doesnt matter anymore

**Comment by githubuser4141** [comment]
I'm not a coder, but IMO shouldn't vampires be able to regenerate their fangs? They are supernatural creatures after all, and it's weird to think you could stop a vampire by removing his/her teeth.

**Comment by PrimeDSS13** [comment]
Maybe, but in-game if someone is getting all (32) of their teeth forcefully removed it's pretty much a guarantee the crew knows they're a vampire and are just doing something a little creative to try and spare the vampire. If they could just re-grow their fangs, it wouldn't really be a route they could take.

File: code/datums/gamemode/role/vampire_role.dm [hunk 2]
```diff
@@ -160,6 +162,10 @@
 		to_chat(M, "<span class='warning'>Remove their mask!</span>")
 		return FALSE
 
+	if(vampire_teeth.amount == 0)
```
Comment (DamianX) [inline code/datums/gamemode/role/vampire_role.dm]: Is `vampire_teeth` guaranteed not to be null here? What about frankensteins, I remember them being weird with butchering drops?

File: code/modules/mob/living/carbon/human/human_attackhand.dm [hunk 3]
```diff
@@ -44,14 +44,22 @@
 		if(2) //Full block
 			damage = 0
 
+	var/attacktype = "bitten"
+	var/datum/butchering_product/teeth/T = locate(/datum/butchering_product/teeth) in M.butchering_drops
+
 	damage = run_armor_absorb(affecting, "melee", damage)
+
+	if(T.amount == 0)
```
Comment (DamianX) [inline code/modules/mob/living/carbon/human/human_attackhand.dm:52]: same here

File: code/modules/mob/living/living_defense.dm [hunk 4]
```diff
@@ -176,7 +176,15 @@
 
 //BITES
 /mob/living/bite_act(mob/living/carbon/human/M as mob)
-	var/damage = rand(1, 5)
+	var/datum/butchering_product/teeth/T = locate(/datum/butchering_product/teeth) in M.butchering_drops
+	var/damage = 0
+	var/attacktype = "bitten"
+
+	if(T.amount > 0)
```
Comment (DamianX) [inline code/modules/mob/living/living_defense.dm]: same here

**Comment by Kurfursten** [comment]
Changes requested

**Comment by PrimeDSS13** [comment]
what should I uh, change

**Comment by Kurfursten** [comment]
Damian was asking you to add handling for a null value of vampire_teeth, or to ensure that it can't become null.

File: code/datums/gamemode/role/vampire_role.dm (hunk 2 above)
Comment (ShiftyRail) [inline code/datums/gamemode/role/vampire_role.dm]:
```suggestion
	if(vampire_teeth?.amount == 0)
```
Should do the trick

**Comment by SonixApache** [comment]
fix it or close it

File: code/modules/mob/living/carbon/human/human_attackhand.dm (hunk 3 above)
Comment (ShiftyRail) [reply code/modules/mob/living/carbon/human/human_attackhand.dm:52]:
```suggestion
	if(T?.amount == 0)
```

File: code/modules/mob/living/living_defense.dm (hunk 4 above)
Comment (ShiftyRail) [reply code/modules/mob/living/living_defense.dm]:
```suggestion
	if(T?.amount > 0)
```

**Pull request merged**